spec/support/shared/examples/msf/db_manager/session.rb

```diff
@@ -117,346 +122,368 @@
+              end
 
-                  session.exploit_datastore['RPORT'] = rport
+              context 'with session.via_exploit' do
+                it 'should create session.via_exploit module' do
+                  framework.modules.should_receive(:create).with(session.via_exploit).and_call_original
 
                   report_session
                 end
 
-                after(:each) do
-                  Timecop.return
+                it 'should create Mdm::Vuln' do
+                  expect {
+                    report_session
+                  }.to change(Mdm::Vuln, :count).by(1)
                 end
 
-                subject(:vuln) do
-                  Mdm::Vuln.last
-                end
+                context 'created Mdm::Vuln' do
+                  let(:mdm_session) do
+                    Mdm::Session.last
+                  end
+
+                  let(:rport) do
+                    nil
+                  end
 
-                it { expect(subject.host).to eq(Mdm::Host.last) }
-                it { expect(subject.refs).to eq([]) }
-                it { expect(subject.exploited_at).to be_within(1.second).of(Time.now.utc) }
+                  before(:each) do
+                    Timecop.freeze
 
-                context "with session.via_exploit 'exploit/multi/handler'" do
-                  context "with session.exploit_datastore['ParentModule']" do
-                    it { expect(subject.info).to eq("Exploited by #{parent_module_fullname} to create Session #{mdm_session.id}") }
-                    it { expect(subject.name).to eq(parent_module_name) }
+                    session.exploit_datastore['RPORT'] = rport
+
+                    report_session
                   end
-                end
 
-                context "without session.via_exploit 'exploit/multi/handler'" do
-                  let(:reference_name) do
-                    'windows/smb/ms08_067_netapi'
+                  after(:each) do
+                    Timecop.return
                   end
 
-                  before(:each) do
-                    path = File.join(
-                        parent_path,
-                        'exploits',
-                        "#{reference_name}.rb"
-                    )
-                    type = 'exploit'
+                  subject(:vuln) do
+                    Mdm::Vuln.last
+                  end
 
-                    # fake cache data for ParentModule so it can be loaded
-                    framework.modules.send(
-                        :module_info_by_path=,
-                        {
-                            path =>
-                                {
-                                    :parent_path => parent_path,
-                                    :reference_name => reference_name,
-                                    :type => type,
-                                }
-                        }
-                    )
+                  it { expect(subject.host).to eq(Mdm::Host.last) }
```

Comment:
If this is just moved code, ignore the recommendation.  It's just in the older style before @lsanchez-r7 corrected us.